### PR TITLE
Prevent mousedown -> drag -> mouseup from creating clicks

### PR DESCRIPTION
**Pull request by omahlama**
Using the mouse, dragging an element causes a click, as the down and up events happen
at the same dom element (which moves under the mouse). This feels very bad and causes
accidental clicks. The fix is to add a click listener to the capture phase and prevent
clicks where:

- the interaction started within react-touch-carousel
- the mouse moved more than 5px

### CODE REVIEW

File: src/touchWithMouseHOC.js [hunk 1]
```diff
@@ -11,24 +11,34 @@ const mockEventTypes = {
   blur: 'touchcancel'
 }
 
+const distance = (x1, y1, x2, y2) =>
+  Math.sqrt(Math.pow(x2-x1, 2) + Math.pow(y2-y1, 2));
+
+const CLICK_TOLERANCE = 5;
```
Comment (xiaody) [inline src/touchWithMouseHOC.js]: This feels magic. It's better we let the user decide the number.
Change the touchWithMouseHOC signature to `touchWithMouseHOC(Component, options)` or `touchWithMouseHOC(options)(Component)`

**Comment by omahlama** [comment]
I added the click tolerance as the second, optional parameter, because most users never have to worry about it, now the don't have to do a `touchWithMouseHOC()(Component)`

**Comment by omahlama** [comment]
So to be more clear, doing it this way keeps users from having to change their code.

**Comment by xiaody** [comment]
Totally understand. But I think it's better to make the second param as an object for forward compatibility.

Also, remember to squash the commits and lint the code. Then it's good for merge.

**Comment by omahlama** [comment]
Working on a project with mandatory semicolons, I just can't help adding them everywhere.

**Comment by omahlama** [comment]
Squashed, linted and using options (because it is definitely a better choice)

**Comment by xiaody** [comment]
Published as `react-touch-carousel@0.8.2`.

**Pull request merged**